packages/report-server/src/reportBuilder/reportBuilder.ts

```diff
@@ -28,7 +30,7 @@ export class ReportBuilder {
   }
 
   public setConfig = (config: Record<string, unknown>) => {
-    this.config = config;
+    this.config = configValidator.validateSync(config);
```

Comment:
I get a type error here but not sure why... I thought it's not worth investing more time at this stage, but happy to look at this together!